lib/lhm.rb

```diff
@@ -49,19 +49,42 @@ def self.change_table(table_name, options = {}, &block)
     true
   end
 
-  def self.cleanup(run = false)
-    lhm_tables = connection.select_values("show tables").select do |name|
-      name =~ /^lhm(a|n)_/
+  def self.cleanup(run = false, options = {})
+    drop_triggers_and_tables('lhm(a|n)_', run, options)
+  end
+
+  def self.cleanup_aborted(run = false, options = {})
+    drop_triggers_and_tables('lhmn_', run, options)
+  end
+
+  def self.drop_triggers_and_tables(pattern, run = false, options = {})
+    lhm_tables = connection.select_values('show tables').select { |name| name =~ /^#{pattern}/ }
+    if options[:until]
+      lhm_tables.select! do |table|
+        table_date_time = Time.strptime(table, 'lhma_%Y_%m_%d_%H_%M_%S')
```

Comment:
Do `lhmn` tables have this timestamp format as well? If so, this pattern will fail.